src/Moq/Mock.cs

```diff
@@ -612,7 +614,11 @@ private static VoidSetupPhrase<T> SetupSetPexProtected<T>(Mock<T> mock, Action<T
 				}
 
 				var setter = last.Invocation.Method;
-				if (!setter.IsPropertySetter())
+				if (setter.ProbablyEventAttachOrDetach())
```

Comment:
In my opinion, this doesn't belong here. It doesn't make any sense to me to have logic for an event `add`/`remove` accessor in a method that is about property  `set`-ters.

Reply:
I'm for adding `SetupAdd` / `SetupRemove`. However I'm not sure how can we avoid them from being delegate-based (but the up-side is that we don't need the other overload - the Expression based overload).

Reply:
Glad to hear you're not set on `SetupSet` for events.

I might try to get some feedback from @kzu on new API methods before making a decision here. I'd like to align Moq 4 with 5 as far as is reasonably possible. If mocking event (un-) subscription isn't planned at all for 5 then I'd find it difficult to add it so late in 4's lifecycle.

We wouldn't be able to avoid delegates if we wanted to continue Moq's current event syntax (`mock => mock.SomeEvent += ...`). That wouldn't be a total no-go but there are some refactorings / code cleanups I'd like to do before adding further static `Setup...` methods.

I'll post back here as soon as I have more news!

Reply:
@idigra: The code cleanups I was referring to are now finally done. Moq now has some new machinery (`ExpressionReconstructor`) that takes a delegate and turns it back into an expression. So it should now be fairly easy to add more delegate-based methods (`SetupAdd`, `SetupRemove`, `VerifyAdd`, `VerifyRemove`).

Just wanted to ping you to check whether you'd still live to work on this at all?

Reply:
Thanks for pinging. I actually found another ways to do what I needed then but maybe I'll find some time of my free time to work on this in the future.

Reply:
@idigra: No worries. :) Should we close this PR here in the meantime?

Reply:
No problem, closed.